src/MinimalDelegation.sol

```diff
@@ -52,12 +52,18 @@ contract MinimalDelegation is IERC7821, ERC1271, EIP712, ERC4337Account, Receive
         if (mode.isBatchedCall()) {
             Call[] calldata calls = executionData.decodeCalls();
             _onlyThis();
-            _dispatch(mode, calls);
+            _dispatch(mode, calls, SignatureUnwrapper.ROOT_KEY_HASH);
         } else if (mode.supportsOpData()) {
-            // executionData.decodeWithOpData();
             (Call[] calldata calls, bytes calldata opData) = executionData.decodeCallsBytes();
-            _authorizeOpData(mode, calls, opData);
-            _dispatch(mode, calls);
+            (uint256 nonce, bytes calldata wrappedSignature) = opData.decodeUint256Bytes();
+            _useNonce(nonce);
```

Comment:
It feels weird to have the top level useNonce function here to me? Usually nonce spending happens right before or in when signature verification is happening 

could we maybe make the flow in this function:

(Call[] memory calls, nonce, signature) = exeuctionData.decodeAllData();

_handleVerifySignature(signedCalls, signature)
_dispatch(mode, calls, keyHash) 

?

Reply:
updated! it's still a bit odd because we need to fully unravel opData to get keyHash